testing/src/main/java/io/stargate/it/http/docsapi/BaseDocumentApiV2Test.java

```diff
@@ -1571,6 +1507,394 @@ public void testPaginationSingleDocSearch() throws IOException {
+    JsonNode nonMatching2 = OBJECT_MAPPER.readTree("{\"value\": \"c\", \"n\": { \"value\": 5}}");
+    RestUtils.put(authToken, collectionPath + "/matching", matching.toString(), 200);
+    RestUtils.put(authToken, collectionPath + "/non-matching", nonMatching.toString(), 200);
+    RestUtils.put(authToken, collectionPath + "/non-matching2", nonMatching2.toString(), 200);
+
+    // Any filter on full collection search should only match the level of nesting of the where
+    // clause
+    String r =
+        RestUtils.get(
+            authToken,
+            hostWithPort
+                + "/v2/namespaces/"
+                + keyspace
+                + "/collections/collection?page-size=2&where={\"value\": {\"$in\": [\"a\", \"b\"]}, \"n.value\": {\"$gt\": 0, \"$lt\": 10}}&raw=true",
+            200);
+
+    String expected = "{\"matching\":{\"value\":\"a\",\"n\":{\"value\":5}}}";
+    assertThat(OBJECT_MAPPER.readTree(r)).isEqualTo(OBJECT_MAPPER.readTree(expected));
+  }
+
+  @Test
+  public void searchPathMatching() throws IOException {
+    JsonNode fullObj1 =
+        OBJECT_MAPPER.readTree("{\"someStuff\": {\"someOtherStuff\": {\"value\": \"a\"}}}");
+    JsonNode fullObj2 = OBJECT_MAPPER.readTree("{\"value\": \"a\"}");
+    RestUtils.put(authToken, collectionPath + "/cool-search-id", fullObj1.toString(), 200);
+    RestUtils.put(authToken, collectionPath + "/cool-search-id-2", fullObj2.toString(), 200);
+
+    String r =
+        RestUtils.get(
+            authToken,
+            hostWithPort
+                + "/v2/namespaces/"
+                + keyspace
+                + "/collections/collection?page-size=2&where={\"someStuff.someOtherStuff.value\": {\"$eq\": \"a\"}}&raw=true",
+            200);
+
+    String expected =
+        "{\"cool-search-id\":{\"someStuff\": {\"someOtherStuff\": {\"value\": \"a\"}}}}";
+    assertThat(OBJECT_MAPPER.readTree(r)).isEqualTo(OBJECT_MAPPER.readTree(expected));
+  }
+
+  @Test
+  public void searchPathSegmentMatching() throws Exception {
+    JsonNode matching1 = OBJECT_MAPPER.readTree("{\"n\": { \"value\": 5}}");
+    JsonNode matching2 = OBJECT_MAPPER.readTree("{\"m\": { \"value\": 8}}");
+    JsonNode nonMatching = OBJECT_MAPPER.readTree("{\"x\": { \"value\": 10}}");
+    RestUtils.put(authToken, collectionPath + "/matching1", matching1.toString(), 200);
+    RestUtils.put(authToken, collectionPath + "/matching2", matching2.toString(), 200);
+    RestUtils.put(authToken, collectionPath + "/non-matching", nonMatching.toString(), 200);
+
+    // Any filter on full collection search should only match the level of nesting of the where
+    // clause
+    String r =
+        RestUtils.get(
+            authToken,
+            hostWithPort
+                + "/v2/namespaces/"
+                + keyspace
+                + "/collections/collection?page-size=3&where={\"n,m.value\": {\"$gte\": 5}}&raw=true",
```

Comment:
unrelated to this PR, but IMHO `,` is a stronger separator than `.` so I personally tend to interpret this as `n` or `m.value`... just my 2 cents :)

Reply:
I think this is valid as it is per json path spec: https://tools.ietf.org/id/draft-goessner-dispatch-jsonpath-00.html#section-1.3-12

Reply:
TIL, thx!

Reply:
JSONPath seems to be different. It requires the alternatives to be wrapped in `[ ]`, does it not? E.g. `["n","m"]["value"]`

Reply:
OK maybe we need to revisit this, not sure if we have someone in the team to help out here? This would be something related to the JS & JSON path maybe.

Reply:
@ashryanbeats : Do you have any suggestions on this ^ (the Q is not specific to this PR and we do not have to fix this immediately)